common/src/main/java/org/keycloak/common/util/PathMatcher.java

```diff
@@ -204,6 +213,13 @@ public String buildUriFromTemplate(String expectedUri, String targetUri) {
             }
 
             if (matchingUri[matchingUri.length - 1] == '\u0000') {
+                if (template.endsWith("*")) {
```

Comment:
Do we want to support here patterns like: "/keycloak-7148/{id}/*/ending". Because at the moment something like this doesn't work as the template doesn't end with the asterisk.